test/integration/inject/inject_test.go

```diff
@@ -53,32 +55,38 @@ func TestInjectManual(t *testing.T) {
 }
 
 func TestInjectManualParams(t *testing.T) {
+
+	injectionValidator := testutil.InjectValidator{
+		DisableIdentity:        true,
+		DisableTap:             true,
+		Version:                "proxy-version",
+		Image:                  "proxy-image",
+		InitImage:              "init-image",
+		ImagePullPolicy:        "Never",
+		ControlPort:            123,
+		SkipInboundPorts:       "234,345",
+		SkipOutboundPorts:      "456,567",
+		InboundPort:            678,
+		AdminPort:              789,
+		OutboundPort:           890,
+		CPURequest:             "10m",
+		MemoryRequest:          "10Mi",
+		CPULimit:               "20m",
+		MemoryLimit:            "20Mi",
+		UID:                    1337,
+		LogLevel:               "warn",
+		EnableExternalProfiles: true,
+	}
+	flags, _ := injectionValidator.GetFlagsAndAnnotations()
+
 	// TODO: test config.linkerd.io/proxy-version
-	cmd := []string{"inject",
+	cmd := append([]string{"inject",
 		"--manual",
 		"--linkerd-namespace=fake-ns",
-		"--disable-identity",
-		"--disable-tap",
 		"--ignore-cluster",
-		"--proxy-version=proxy-version",
-		"--proxy-image=proxy-image",
-		"--init-image=init-image",
-		"--image-pull-policy=Never",
-		"--control-port=123",
-		"--skip-inbound-ports=234,345",
-		"--skip-outbound-ports=456,567",
-		"--inbound-port=678",
-		"--admin-port=789",
-		"--outbound-port=890",
-		"--proxy-cpu-request=10m",
-		"--proxy-memory-request=10Mi",
-		"--proxy-cpu-limit=20m",
-		"--proxy-memory-limit=20Mi",
-		"--proxy-uid=1337",
-		"--proxy-log-level=warn",
-		"--enable-external-profiles",
-		"testdata/inject_test.yaml",
-	}
+	}, flags...)
+
+	cmd = append(cmd, "testdata/inject_test.yaml")
 
 	out, stderr, err := TestHelper.LinkerdRun(cmd...)
```

Comment:
I believe there's a way to transform this `out` into a k8s object, from which you could extract `v1.Pod` and use `injectionValidator.ValidatePod()` on it like you do below, so we can get rid of the golden file comparison?

Reply:
Yes I will attempt that. Very good idea. Not sure why my CI is failing. Seems completely unrelated...